mode/clike/clike.js

```diff
@@ -519,6 +519,16 @@ CodeMirror.defineMode("clike", function(config, parserConfig) {
         stream.eatWhile(/[\w\$_]/);
         return "meta";
       },
+      "{": function(stream, state) {
+        // match a single full line with params of multi-line anonymous fn, e.g.:
+        // { a =>\n
+        // { (aa: Int, bb: Int) =>\n
+        var singleLineParams = /^(\w|[:.,() ])+=>$/;
+        if (!stream.match(singleLineParams, true)) // eat the line, if matches
```

Comment:
This'll disable all highlighting for the parameters. That seems undesirable. Also, I believe Scala allows default values for parameters, so the actual syntax for a parameter list may contain arbitrary subexpressions, which I assume your regexp won't cover. Would it make more sense to delay the context manipulation until reaching the `=>`?